src/Illuminate/Database/Query/Grammars/Grammar.php

```diff
@@ -612,6 +612,20 @@ protected function compileUnion(array $union)
     }
 
     /**
+     * Compile an exists statement into SQL.
+     *
+     * @param \Illuminate\Database\Query\Builder $query
+     *
+     * @return string
+     */
+    public function compileExists(Builder $query)
+    {
+        $select = $this->compileSelect($query);
```

Comment:
Should the query be mutated to change the columns so that a `select *` is not generated?

Eg 
```
$query->columns(1);
```
Should generate 
```
select exists(select 1 from "users") as `exists`
```
in the test